packages/noports_core/lib/src/sshrvd/socket_connector.dart

```diff
@@ -1,48 +1,123 @@
-void socketConnector(ConnectorParams params) async {
-  var (sendPort, portA, portB, session, forAtsign, snoop) = params;
+void socketConnector(ConnectorParams connectorParams) async {
+  var (
+    sendPort,
+    portA,
+    portB,
+    sshrvdSessionParamsJsonString,
+    snoop,
+    verbose,
+  ) = connectorParams;
+
+  if (verbose) {
+    AtSignLogger.root_level = 'INFO';
+  } else {
+    AtSignLogger.root_level = 'WARNING';
+  }
+
+  final logger = AtSignLogger(' sshrvd / socket_connector ');
 
-  logger.info('Starting socket connector session $session for $forAtsign');
+  SshrvdSessionParams sshrvdSessionParams =
+      SshrvdSessionParams.fromJson(jsonDecode(sshrvdSessionParamsJsonString));
+  logger.info(
+      'Starting socket connector session for ${sshrvdSessionParams.toJson()}');
+
+  /// Create the socketAuthVerifiers as required
+  Map expectedPayloadForSignature = {
+    'sessionId': sshrvdSessionParams.sessionId,
+    'clientNonce': sshrvdSessionParams.clientNonce,
+    'rvdNonce': sshrvdSessionParams.rvdNonce,
+  };
+
+  SocketAuthVerifier? socketAuthVerifierA;
+  if (sshrvdSessionParams.authenticateSocketA) {
+    String? pkAtSignA = sshrvdSessionParams.publicKeyA;
+    if (pkAtSignA == null) {
+      logger.shout('Cannot spawn socket connector.'
+          ' Authenticator for ${sshrvdSessionParams.atSignA}'
+          ' could not be created as PublicKey could not be'
+          ' fetched from the atServer.');
+      throw Exception('Failed to create SocketAuthenticator'
+          ' for ${sshrvdSessionParams.atSignA} due to failure to get public key for ${sshrvdSessionParams.atSignA}');
+    }
+    socketAuthVerifierA = SignatureAuthVerifier(
+      pkAtSignA,
+      jsonEncode(expectedPayloadForSignature),
+      sshrvdSessionParams.rvdNonce!,
+      sshrvdSessionParams.atSignA,
+    );
+  }
+
+  SocketAuthVerifier? socketAuthVerifierB;
+  if (sshrvdSessionParams.authenticateSocketB) {
+    String? pkAtSignB = sshrvdSessionParams.publicKeyB;
+    if (pkAtSignB == null) {
+      logger.shout('Cannot spawn socket connector.'
+          ' Authenticator for ${sshrvdSessionParams.atSignB}'
+          ' could not be created as PublicKey could not be'
+          ' fetched from the atServer');
+      throw Exception('Failed to create SocketAuthenticator'
+          ' for ${sshrvdSessionParams.atSignB} due to failure to get public key for ${sshrvdSessionParams.atSignB}');
+    }
+    socketAuthVerifierB = SignatureAuthVerifier(
+      pkAtSignB,
+      jsonEncode(expectedPayloadForSignature),
+      sshrvdSessionParams.rvdNonce!,
+      sshrvdSessionParams.atSignB!,
+    );
+  }
 
   /// Create the socket connector
-  SocketConnector socketStream = await SocketConnector.serverToServer(
-    serverAddressA: InternetAddress.anyIPv4,
-    serverAddressB: InternetAddress.anyIPv4,
-    serverPortA: portA,
-    serverPortB: portB,
-    verbose: snoop,
-  );
+  SocketConnector connector = await SocketConnector.serverToServer(
```

Comment:
Renamed here (and other places) to 'connector'; the name `socketStream` was confusing me since SocketConnector is not a stream of Sockets